src/app/core/auth/auth.service.ts

```diff
@@ -116,21 +125,24 @@ export class AuthService {
    * Returns the authenticated user
    * @returns {User}
    */
-  public authenticatedUser(token: AuthTokenInfo): Observable<Eperson> {
+  public authenticatedUser(token: AuthTokenInfo): Observable<EPerson> {
     // Determine if the user has an existing auth session on the server
     const options: HttpOptions = Object.create({});
     let headers = new HttpHeaders();
     headers = headers.append('Accept', 'application/json');
     headers = headers.append('Authorization', `Bearer ${token.accessToken}`);
     options.headers = headers;
-    return this.authRequestService.getRequest('status', options)
-      .map((status: AuthStatus) => {
+    return this.authRequestService.getRequest('status', options).pipe(
+      switchMap((status: AuthStatus) => {
+
         if (status.authenticated) {
-          return status.eperson[0];
+          // TODO this should be cleaned up, AuthStatus could be parsed by the RemoteDataService as a whole...
```

Comment:
Should this TODO be turned into a task/ticket for the Angular UI?  I just want to be sure that this task isn't lost/forgotten.

Reply:
I think we will be able to clean this up as well after [DS-4006](https://jira.duraspace.org/browse/DS-4006) has a fix. EPersons in authentication status responses cannot be parsed as expected because of the way the response is set up at the moment. 

I created an [issue](https://github.com/DSpace/dspace-angular/issues/292) for the Angular UI as well.